app/controllers/concerns/api/articles_controller.rb

```diff
@@ -0,0 +1,136 @@
+module Api
+  module ArticlesController
+    extend ActiveSupport::Concern
```

Comment:
To avoid duplicate controller code (that supports both `V0` and `V1`) the entire controller now lives in a concern that gets included in both controllers. I feel like this is the least impactful way to do this, otherwise we'll have two identical controllers in each version during the transition period. Always open for feedback here, i.e. alternatives or possible ways to make this a cleaner implementation.